src/js/actions/libraries.js

```diff
@@ -821,7 +827,13 @@ define(function (require, exports) {
                         // So, although the command does not resolve until the placement has finished,
                         // it does resolve before the layer model has been updated because the event
                         // has not yet been received. For more details see issue #2177.
-                        return descriptor.playObject(placeObj, { synchronous: false });
+                        return descriptor.playObject(placeObj, {
+                            synchronous: false,
+                            historyStateInfo: {
```

Comment:
I'm not convinced that this is doing what you are expecting.  When I drag a graphic asset onto canvas, I see two history states.  Same behavior on master as this PR branch.
~[ss](http://cl.ly/image/2K1G2z333f0k)

I think there is a separate issue here related to that first history state for the "wrong" document, btw.

Am I missing something?  Here is a vid: http://cl.ly/1o0O3g0g3Z0E